config/ompi_check_pvfs2.m4

```diff
@@ -46,70 +42,27 @@ AC_DEFUN([OMPI_CHECK_PVFS2],[
              [Build Pvfs2 support, optionally adding DIR/include, DIR/lib, and DIR/lib64 to the search path for headers and libraries])])
     OPAL_CHECK_WITHDIR([pvfs2], [$with_pvfs2], [include/pvfs2.h])
 
-    AC_ARG_WITH([pvfs2-libs],
-        [AC_HELP_STRING([--with-pvfs2-libs=LIBS],
-                       [Libraries to link with for pvfs2])])
-
-    temp_with_pvfs2="$with_pvfs2"
     AS_IF([test -z "$with_pvfs2"],
-          [with_pvfs2="/usr/local"])
+          [ompi_check_pvfs2_dir="/usr/local"],
+          [ompi_check_pvfs2_dir=$with_pvfs2])
 
-    temp_with_pvfs2_libs="$with_pvfs2_libs"
-    AS_IF([test -z "$with_pvfs2_libs"],
-	[with_pvfs2_libs="pvfs2 pthread"])
+    if test -e "$ompi_check_pvfs2_dir/lib64" ; then
+        ompi_check_pvfs2_libdir="$ompi_check_pvfs2_dir/lib64"
+    else
+        ompi_check_pvfs2_libdir="$ompi_check_pvfs2_dir/lib"
+    fi
 
     # Add correct -I and -L flags
-    AS_IF([test -d "$with_pvfs2/include"],
-        [check_pvfs2_CPPFLAGS="-I$with_pvfs2/include"
-            $1_CPPFLAGS="$check_pvfs2_CPPFLAGS"
-            CPPFLAGS="$CPPFLAGS $check_pvfs2_CPPFLAGS"],
-	[ompi_check_pvfs2_happy="no"])
+    OPAL_CHECK_PACKAGE([$1], [pvfs2.h], [pvfs2], [PVFS_util_resolve], [$PTHEAD_LIBS],
```

Comment:
Mmm... good point; we may have filled in the pthreads flags into the main LIBS, CFLAGS, etc. already.  In which case, you're right: you should just pass `[]` for that argument.

Can you take a quick pass at verifying that we do actually do that?  E.g., in opal_config_pthread.m4, right before line 651, do something like:

```sh
PTHREAD_CPPFLAGS="-DEDGAR_IS_AWESOME_CPPFLAGS $PTHREAD_CPPFLAGS"
PTHREAD_LIBS="-DEDGAR_IS_AWESOME_LIBS $PTHREAD_LIBS"
...etc.
```

Then you can see if those (harmless) -D's make it through to CFLAGS, LIBS, etc. by the time we're down in your component's configure.m4.